dhis2/dhis2-core
# refactor: system and DB info as immutable objects

### Summary
The original goal was to add the DB time info to the system info so it can be considered for time-zone related issues.
The main changes were needed to make the added information be dynamic so the time is not frozen since startup.

The mutable info objects were hard to understand so the main changes become to make them into immutable objects instead.
The `DatabaseInfo` is no longer a singleton bean and must always be obtained from the `DatabaseInfoProvider` as it contains fields that are not stable. 

Also integration tests now use the actual `SystemService` implementation not a mocked one.

### Automatic Testing
Added 2 scenarios that fetch the system info, one with and one without super access.
Added a unit test to check all sensitive information are cleared.

### Manual Testing
* check `/api/system/info` for new property database `time`
* check pressing F5 (refresh) does result in a updated `time` value
